### PR TITLE
feat(themes): full theming example with bootstrap

### LINKED ISSUE #2603: HowTos: Full theming example with a bootstrap
Create an article, which walks users through how to configure the parameters for the Ignite UI for Angular bootstrap mixin, so the components theme fits nicely with their general application bootstrap theme.

### PR DESCRIPTION
**Pull request by SisIvanova**
Closes #2603 

### Checklist:
 
 - [x] check topic's TOC/menu and paragraph headings
 - [x] link to other topics using `../relative/path.md`
 - [x] at the References section at the end of the topic add links to topics, samples, etc
 - [x] reference API documentation instead of adding a section with API

<br />

 - [ ] use valid component names - [Data] Grid, `IgxSelectComponent`, `<igx-combo>`
 - [ ] use spell checker tool (VS Code, Grammarly, Microsoft Editor)
 - [ ] add inline `code blocks` for the names of classes / tags / properties
 - [ ] add language descriptor for the ```code blocks```
 - [ ] check broken links (use browser add-on)
 - [ ] check if sample is working and fully visible in the topic
 - [ ] check if sample is working and fully visible in the StackBlitz
 - [ ] check if code blocks match the code in StackBlitz demo

<br />

 - [ ] follow SEO guidelines to fill topic's metadata ([SEO guidelines](https://infragisticsinc297.sharepoint.com/:w:/g/Groups/marketing/EWz9InT4FDlErHCumxsKGY4Bd8H03yhRWxDFk47luRz-_Q?e=S5wWcx))

<br />

 - [ ] do not resolve requested changes (leave that to the reviewer)
 - [ ] add `pending-localization` label when the review of the PR is done
 - [ ] add a member from the localization team to translate it

### CODE REVIEW

File: en/components/themes/misc/bootstrap-theming.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+@NgModule({
+    imports: [
+        ...,
+        NgbModule,
+        ...
+    ]
+)}
+```
+
+At this point your applications is ready to use the NG Bootstrap components. You can find more information about using Bootstrap library at their [`official documentation`](https://ng-bootstrap.github.io/#/getting-started).
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:48]:
```suggestion
At this point your applications is ready to use the NG Bootstrap components. You can find more information about using the Bootstrap library at their [`official documentation`](https://ng-bootstrap.github.io/#/getting-started).
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+### Add Ignite UI for Angular 
+
+To install the Ignite UI for Angular package, along with all of its dependencies, font imports and styles references to your project, run the following command:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:52]:
```suggestion
To install the Ignite UI for Angular package along with all of its dependencies, font imports, and styles references, run the following command in your project,: 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+Clicking on the `More` buttons, you will see the [`igx-dialog`]({environment:angularApiUrl}/classes/igxdialogcomponent.html):
+
+<img src="../../../images/igx-dialog.png" width="100%">
+
+Next, we have added an [`ngb-accordion`](https://ng-bootstrap.github.io/#/components/accordion/examples) showing information about credit cards. Inside its content, there is an [`igx-list`]({environment:angularApiUrl}/classes/igxlistcomponent.html) and `igx-button`.
+
+<img src="../../../images/ngb-accordion.png" width="100%">
+
+Finally, we inserted an igx `icon button` in the top right corner, that changes the color palette for the whole app:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:89]:
```suggestion
Finally, we inserted an Ignite UI for Angular `icon button` in the top right corner, that changes the theme of the whole app: 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+```scss
+// _variables.scss
+
+@import '~igniteui-angular/lib/core/styles/themes/index';
+``` 
+
+### Palettes
+
+The Bootstrap library makes use of a sass map called [`$theme-colors`](https://getbootstrap.com/docs/4.0/getting-started/theming/#theme-colors) to provide a palette composed of eight colors:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:105]:
```suggestion
The Bootstrap library makes use of a Sass map called [`$theme-colors`](https://getbootstrap.com/docs/4.0/getting-started/theming/#theme-colors) to provide a palette composed of eight colors:
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+    "success":    $success,
+    "info":       $info,
+    "warning":    $warning,
+    "danger":     $danger,
+    "light":      $light,
+    "dark":       $dark
+);
+```
+
+On the other hand, the Ignite UI [`igx-palette`]({environment:sassApiUrl}/index.html#function-igx-palette) function generates a color palette map including `primary`, `secondary`, `grays`, `info`, `success`, `warn`, and `error` colors and their 74 color variants. Our predefined bootstrap palette in turn consists of seven colors:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:120]:
```suggestion
Ignite UI for Angular's [`igx-palette`]({environment:sassApiUrl}/index.html#function-igx-palette) function generates a color palette map including `primary`, `secondary`, `grays`, `info`, `success`, `warn`, and `error` colors and their color variants. Our predefined bootstrap palette in turn consists of seven colors: 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+    $secondary: #6c757d,
+    $info: #17a2b8,
+    $success: #28a745,
+    $warn: #ffc107,
+    $error: #dc3545,
+    $surface: #f8f9fa
+);
+```
+
+As you can see most of the colors in their palette overlap the colors in the palettes of Ignite UI. Thus, we can simply map the Bootstrap theme colors with our light or dark bootstrap palette colors.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:134]:
```suggestion
As you can see most of the colors in the Bootstrap palette overlap with the colors defined the Bootstrap palette of Ignite UI for Angular. Hence, we can simply map the Bootstrap theme colors to our light or dark bootstrap palette colors. 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+    $success: #28a745,
+    $warn: #ffc107,
+    $error: #dc3545,
+    $surface: #f8f9fa
+);
+```
+
+As you can see most of the colors in their palette overlap the colors in the palettes of Ignite UI. Thus, we can simply map the Bootstrap theme colors with our light or dark bootstrap palette colors. 
+
+First, we are going to define Sass variables for the [`$light-bootstrap-palette`]({environment:sassApiUrl}/index.html#variable-light-bootstrap-palette) colors:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:136]:
```suggestion
First, we are going to define Sass variables that extract values from the [`$light-bootstrap-palette`]({environment:sassApiUrl}/index.html#variable-light-bootstrap-palette):
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+```scss
+// Defining custom color palette
+$custom-dark-palette: igx-palette(
+    $primary: #ecaa53,
+    $secondary: #011627,
+    $grays: #fff,
+    $surface: #222
+);
+
+// Creating Sass variables for its primary and secondary colors
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:159]:
```suggestion
// Creating Sass variables for primary and secondary colors
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+$dark-primary: igx-color($custom-dark-palette, "primary");
+$dark-secondary: igx-color($custom-dark-palette, "secondary");
+```
+
+>[!NOTE]
+>Visit our [`palettes with sass`](../sass/palettes.md) section to discover more about the palettes provided by Ignite UI for Angular and learn how to create a new one.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:165]:
```suggestion
>Visit our [`palettes with Sass`](../sass/palettes.md) section to discover more about the palettes provided by Ignite UI for Angular and learn how to create a new one.
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+
+### Themes
+
+In order to switch between `light` and `dark` mode, we are adding a custom class to the `host` element which will be changed on button click event. In our stylesheet file, we are going to include different color palettes according to that class.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:169]:
```suggestion
In order to switch between `light` and `dark` mode, we are adding a custom class to the `host` element which will be changed on button click. In our stylesheet file, we are going to include different color palettes scoped to each class.
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+
+#### Light mode
+
+The Ignite UI for Angular comes in with a predefined theme inspired by [Bootstrap 4](https://getbootstrap.com/) library. To use it, first, you have to include our `igx-core` mixin and then our built-in theme mixin - [igx-bootstrap-light-theme]({environment:sassApiUrl}/index.html#mixin-igx-bootstrap-light-theme). We will also make use of our predefined bootstrap palette - [$light-bootstrap-palette]({environment:sassApiUrl}/index.html#variable-light-bootstrap-palette).
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:173]:
```suggestion
Ignite UI for Angular comes with predefined themes inspired by the [Bootstrap 4](https://getbootstrap.com/) library. To use them, first, you have to include our `igx-core` mixin and then our built-in theme mixin - [igx-bootstrap-light-theme]({environment:sassApiUrl}/index.html#mixin-igx-bootstrap-light-theme). We will also make use of our predefined bootstrap palette - [$light-bootstrap-palette]({environment:sassApiUrl}/index.html#variable-light-bootstrap-palette). 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+
+In order to switch between `light` and `dark` mode, we are adding a custom class to the `host` element which will be changed on button click event. In our stylesheet file, we are going to include different color palettes according to that class.
+
+#### Light mode
+
+The Ignite UI for Angular comes in with a predefined theme inspired by [Bootstrap 4](https://getbootstrap.com/) library. To use it, first, you have to include our `igx-core` mixin and then our built-in theme mixin - [igx-bootstrap-light-theme]({environment:sassApiUrl}/index.html#mixin-igx-bootstrap-light-theme). We will also make use of our predefined bootstrap palette - [$light-bootstrap-palette]({environment:sassApiUrl}/index.html#variable-light-bootstrap-palette). 
+
+The background color for our application needs to be set explicitly on the host element. In our sample, we want to use the `surface` color of the passed palette.
+
+Along with including the themes, at this point, we have to modify the bootstrap `$theme-colors` map with our sass variables created earlier.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:177]:
```suggestion
At this point we have to modify the Bootstrap `$theme-colors` map with the Sass variables we created earlier.
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 13]
```diff
@@ -0,0 +1,352 @@
...
+Lastly, we need to import the bootstrap library - *always import it at the end!*
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:178]:
```suggestion
Lastly, we need to import the Bootstrap library - *always import it at the end!*
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 14]
```diff
@@ -0,0 +1,352 @@
...
+            $theme-colors: (
+                "primary": $light-primary,
+                "secondary": $light-secondary,
+                "success": $light-success,
+                "info": $light-info,
+                "warning": $light-warning,
+                "danger": $light-danger
+            );
+
+            // Importing Bootstrap SCSS file
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:202]:
```suggestion
            // Importing Bootstrap .scss file
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 15]
```diff
@@ -0,0 +1,352 @@
...
+                "dark": $custom-dark,
+            );
+        }
+    }
+}
+```
+
+#### Dark mode
+
+For our dark variant of the application we are going to use our newly created `$custom-dark-palette` so we have to include it in the `dark class` styles and also modify the `$theme-colors` map with the new values.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:230]:
```suggestion
For our dark variant, we are going to use our newly created `$custom-dark-palette`. We have to include it in the `dark` class styles and also modify the `$theme-colors` map with the new values. 
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 16]
```diff
@@ -0,0 +1,352 @@
...
+        }
+    }
+}
+```
+
+#### Dark mode
+
+For our dark variant of the application we are going to use our newly created `$custom-dark-palette` so we have to include it in the `dark class` styles and also modify the `$theme-colors` map with the new values. 
+
+All the colors in the Ignite UI components come from the passed palette, therefore they fit nicely in the dark mode without any additional adjustments. However, we have to do some more styling changes for the ng-bootstrap components:
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:232]:
```suggestion
All components in Ignite UI for Angular use colors from the passed palette, therefore they fit nicely in the dark mode without any additional adjustments. However, we have to do some more styling changes for the ng-bootstrap components:
```

File: en/components/themes/misc/bootstrap-theming.md [hunk 17]
```diff
@@ -0,0 +1,352 @@
...
+        );
+    }
+}
+```
+
+Then, add a CSS class to your navbar component following the pattern "bg - color from the palette - color variant". In our sample app, we are using `bg-grays-200`.
+
+### Typography
+
+Ignite UI for Angular exposes four default type scales for each of its themes, which can be used inside the [`igx-typography`]({environment:sassApiUrl}/index.html#mixin-igx-typography) mixin to define the global typography styles for an application. In our example, we are going to apply the bootstrap predifined `typeface` and `type-scale` but you can create your custom ones if you wish.
```
Comment (simeonoff) [inline en/components/themes/misc/bootstrap-theming.md:318]:
```suggestion
Ignite UI for Angular exposes four default type scales for each of its themes, which can be used inside the [`igx-typography`]({environment:sassApiUrl}/index.html#mixin-igx-typography) mixin to define the global typography styles of an application. In our example, we are going to apply the bootstrap predifined `typeface` and `type-scale` but you can create custom ones if you wish. 
```

**Review by simeonoff — CHANGES_REQUESTED** [review]

**Review by simeonoff — APPROVED** [review]

**Comment by StefanIvanov** [comment]
@aatanasov2 will help us with the SEO since this topic can draw lots of traffic to the docs and samples :)

**Pull request merged**